src/components/Wizards/CreateVmWizard/tests/NetworksTab.test.js

```diff
@@ -1,13 +1,116 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { shallow, mount } from 'enzyme';
 import { NetworksTab } from '..';
 import NetworksTabFixture from '../fixtures/NetworksTab.fixture';
+import { TableFactory } from '../../../Table/TableFactory';
+import { POD_NETWORK } from '../../../../constants';
 
-const testNetworksTab = () => <NetworksTab {...NetworksTabFixture.props} />;
+const testNetworksTab = (props = NetworksTabFixture.props, mnt = false) => {
+  const networksTab = <NetworksTab {...props} />;
+  return mnt ? mount(networksTab) : shallow(networksTab);
+};
+
+const defaultNetworks = [
+  { id: 1, isBootable: false, name: 'nic1', mac: 'mac1', network: POD_NETWORK, errors: [] },
+  { id: 2, isBootable: true, name: 'nic2', mac: 'mac2', network: 'network2', errors: [] },
+  { id: 3, isBootable: false, name: 'nic3', mac: 'mac3', network: 'network3', errors: [] }
+];
 
 describe('<NetworksTab />', () => {
   it('renders correctly', () => {
-    const component = shallow(testNetworksTab());
+    const component = testNetworksTab();
     expect(component).toMatchSnapshot();
   });
+  it('passes existing networks to FormFactory', () => {
+    const props = { ...NetworksTabFixture.props };
+    props.networks = defaultNetworks;
+    const component = testNetworksTab(props);
+
+    const passedRows = component.find(TableFactory).props().rows;
+    expect(passedRows).toHaveLength(3);
+    expect(passedRows[0].addendum).toBeNull();
+    expect(passedRows[0].renderConfig).toEqual(1);
+    expect(passedRows[1].addendum).toEqual('(Bootable)');
+    expect(passedRows[1].isBootable).toBeTruthy();
+    expect(passedRows[1].renderConfig).toEqual(0);
+    expect(passedRows[2].addendum).toBeNull();
+    expect(passedRows[2].isBootable).toBeFalsy();
+    expect(passedRows[2].renderConfig).toEqual(0);
+  });
+  it('checkPxeBootNetwork', () => {
+    const props = { ...NetworksTabFixture.props };
+    props.networks = defaultNetworks;
+    const component = testNetworksTab(props);
+
+    const { rows } = component.state();
+    expect(component.instance().checkPxeBootNetwork(rows)).toBeFalsy();
```

Comment:
I think it is better to put the function outside of the component and test it separately

Reply:
Agree. Will help readability